cmake/DetectOptions.cmake

```diff
@@ -137,13 +138,32 @@ if(ZFP_FOUND)
       return !zfp_stream_set_execution(stream, zfp_exec_cuda);
     }"
     ADIOS2_HAVE_ZFP_CUDA)
+
```

Comment:
IS this needed, we can check if zfp version is greater than 0.5.4 to check that we support 4D functions. 

https://github.com/LLNL/zfp/commit/83402a16fb71106d44fe6253195fb9ec03728e4e

Reply:
Commit 9b75283 solves it through checking the version which involves a cumbersome preprocessor if clause. So checking for feature availability seemed to be the option that would not clutter the source file with version checking. Certainly, a rollback to 9b75283 is possible.

Reply:
@gregorweiss it appears to me that the https://github.com/ornladios/ADIOS2/commit/9b75283a15d72ac353f908591ba91ce36d139e5a approach is takes less LOC than this and it is a stronger alternative. 

> that would not clutter the source file with version checking

Cmake code is also source code

The rationale for it to be a better alternative is that:
 - It is faster, no need to build/run test in the configure step.
 - The build/run step could fail for other reasons unrelated to whether it supports 4d arrays.

Finally, the reason of using a build/run test for checking if a feature of a lib is available/running is the last resort that we have when we have no other way of checking this directly in the source code.

Reply:
@vicentebolea Thank you for sharing your train of thought. Rollback is ready. There was also a flaw that I fixed in the if-condition of commit 9b75283. The new commit should check that version number is greater than 0.5.3.

Reply:
looks good to me, please fix format and Ill approve this. Many thanks!